tensorflow/core/kernels/transpose_op.cc

```diff
@@ -133,7 +133,16 @@ void TransposeOp::Compute(OpKernelContext* ctx) {
   OP_REQUIRES(ctx, TensorShapeUtils::IsVector(perm.shape()),
               errors::InvalidArgument("perm must be rank 1, got shape ",
                                       perm.shape().DebugString()));
-
+  // Validation for negative values inside `perm`.
+  auto perm_vector = perm.vec<int>();
+  for (int i = 0; i < perm.dim_size(0); i++) {
+    if (0 > perm_vector(i)) {
```

Comment:
This is a little difficult to read would be better as perm_vector(i) < 0

Reply:
Sure will do changes.
But I have a small confusion here. Whether `i < perm.dim_size(0)` should be `i < perm.shape().dim_size(0)` in for loop ?

Reply:
@sagunb

Reply:
It should be `parm.dim_size(0)` since you are iterating on `perm_vector` (which is the 1D representation of the tensor, hence why you only care about the first dim size. I actually recommend switching this to `perm.NumElements()` to make it more readable (and save it in a variable, so you don't keep calling it in the loop).

The `perm.shape().dim_size(0)` takes the `perm` tensor, builds a new tensor for its shape and then looks at the first dimension (which is exactly the number of elements of the shape, that is the rank of the `perm` tensor):

```python
> perm = [1,2,3]
> perm.dim_size(0)
3
> perm.shape().dim_size(0)
1
> perm = [[1,2,3]]
> perm.dim_size(0)
1
> perm.shape().dim_size(0)
1
```

Reply:
@mihaimaruseac , Thank you very much for the reply.